### PR TITLE
Update documentation regarding pg_upgrade

### PR DESCRIPTION
**Pull request by mkindahl**
The variable `timescaledb.restoring` does not need to be set when
performing a `pg_upgrade` so that was removed from the instructions.

In addition, the `timescaledb_post_restore` function will now *reset*
the `timescaledb.restoring` flag rather than setting it to `off`.
Having `timescaledb.restoring` set to `off` on a database can cause
issues when upgrading PostgreSQL for older versions of TimescaleDB.

### CODE REVIEW

File: api.md [hunk 1]
```diff
@@ -2767,6 +2767,10 @@ Specifically this sets the `timescaledb.restoring` GUC to `on` and stops any
 background workers which may have been performing tasks until the [`timescaledb_post_restore`](#timescaledb_post_restore)
 fuction is run following the restore. See [backup/restore docs][backup-restore] for more information.
 
+Note that you do not need so use this function when doing an upgrade
```
Comment (svenklemm) [inline api.md:2770]: should this be tip or warning?

Comment (mkindahl) [reply api.md]: I think a warning is more appropriate. Changed.

File: api.md [hunk 2]
```diff
@@ -2767,6 +2767,10 @@ Specifically this sets the `timescaledb.restoring` GUC to `on` and stops any
 background workers which may have been performing tasks until the [`timescaledb_post_restore`](#timescaledb_post_restore)
 fuction is run following the restore. See [backup/restore docs][backup-restore] for more information.
 
+>:WARNING: Note that you do not need so use this function when doing an upgrade
```
Comment (svenklemm) [inline api.md:2770]: there is typo `need so` => `need to`

File: api.md [hunk 3]
```diff
@@ -2767,6 +2767,10 @@ Specifically this sets the `timescaledb.restoring` GUC to `on` and stops any
 background workers which may have been performing tasks until the [`timescaledb_post_restore`](#timescaledb_post_restore)
 fuction is run following the restore. See [backup/restore docs][backup-restore] for more information.
 
+>:WARNING: Note that you do not need to use this function when doing an upgrade
+using `pg_upgrade` and using this function when doing an upgrade could
+cause issue in previous versions of TimescaleDB.
```
Comment (davidkohn88) [inline api.md:2772]: `using this function when doing an upgrade could cause issue in previous versions of TimescaleDB.` 

I would reword as using this function when doing an upgrade can cause problems in previous versions....

Comment (mkindahl) [reply api.md]: Reformulated.

File: api.md [hunk 4]
```diff
@@ -2775,6 +2775,8 @@ Specifically this sets the `timescaledb.restoring` GUC to `on` and stops any
 background workers which may have been performing tasks until the [`timescaledb_post_restore`](#timescaledb_post_restore)
 fuction is run following the restore. See [backup/restore docs][backup-restore] for more information.
 
+>:WARNING: Using this function when doing an upgrade could cause issue in previous versions of TimescaleDB.
```
Comment (davidkohn88) [inline api.md:2778]: Nit: could cause an issue, or could cause issues, not could cause issue.

Comment (mkindahl) [reply api.md:2778]: Used "could cause issues".

File: api.md [hunk 5]
```diff
@@ -2775,6 +2775,8 @@ Specifically this sets the `timescaledb.restoring` GUC to `on` and stops any
 background workers which may have been performing tasks until the [`timescaledb_post_restore`](#timescaledb_post_restore)
 fuction is run following the restore. See [backup/restore docs][backup-restore] for more information.
 
+>:WARNING: Using this function when doing an upgrade could cause issues in previous versions of TimescaleDB.
```
Comment (mfreed) [inline api.md:2778]: What does  "previous versions" mean?  Should we specify anything like "versions earlier than 1.x"?

Comment (mkindahl) [reply api.md:2778]: Clarified.

File: api.md (hunk 2 above)
Comment (mkindahl) [reply api.md]: Fixed.

File: api.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkindahl) [reply api.md]: Nothing special is needed, strangely enough.
